Mathlib/Data/Set/Intervals/Basic.lean

```diff
@@ -840,6 +840,16 @@ theorem Icc_diff_Ioo_same (h : a ≤ b) : Icc a b \ Ioo a b = {a, b} := by
   simp [insert_subset_iff, h]
 #align set.Icc_diff_Ioo_same Set.Icc_diff_Ioo_same
 
+theorem Icc_diff_Ioo_same' (h : a ≤ b) : (Icc a b \ {a, b}) ∪ {a, b} = Icc a b := by
+  refine diff_union_of_subset ?h
+  intro x hx
+  cases hx with
+  | inl h2 => rw [h2]; exact left_mem_Icc.mpr h
+  | inr h2 => rw [h2]; exact right_mem_Icc.mpr h
+
+theorem Icc_diff_Ioo_same'' (h : a ≤ b) : Ioo a b ∪ {a, b} = Icc a b := by
```

Comment:
This should be called `Ioo_union_both` to match `Ioo_union_left`

Reply:
(There should also be an easy proof that goes via `Ioo_union_left` and `Ico_union_right`)

Reply:
Done. 
However, I cannot use `Ioo_union_left` here, because I cannot deduce that `a < b`.​